.github/workflows/build-push-docker-image.yml

```
@@ -0,0 +1,36 @@
+name: Publish Docker image
+
+on:
+  release:
+    types: [published]
+
+jobs:
+  build_and_push_docker_image:
+    name: Push Docker image to Docker Hub
+    if: github.repository_owner == 'NethermindEth'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v3
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Log in to Docker Hub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+
+      - name: Get release version
+        id: get_version
+        shell: bash
+        run: echo "TAG_FROM_REPO=$(git describe --tags --always | cut -d- -f1)" >> $GITHUB_OUTPUT
+
+      - name: Build and push Docker image
+        uses: docker/build-push-action@v4
+        with:
+          context: .
+          file: ./Dockerfile
+          push: true
+          tags: nethermind/warp:${{ steps.get_version.outputs.TAG_FROM_REPO }}
```

I think we don't need the release version. Do you think we could use `tag_name`?
```suggestion
      - name: Build and push Docker image
        uses: docker/build-push-action@v4
        with:
          context: .
          file: ./Dockerfile
          push: true
          tags: nethermind/warp:${{ github.event.release.tag_name }}
```

---

Is it possible that `tag_name` is an empty string? I thought to get the release version to have a different image after every new release

---

`tag_name` should be whatever you set when creating a release in github UI. I don't think it can be empty, it wouldn't make sense to create a release with a blank version number